makefiles/vars.inc.mk

```diff
@@ -19,6 +19,7 @@ export MCU                   # The MCU, set by the board's Makefile.include, or
 export INCLUDES              # The extra include paths, set by the various Makefile.include files.
 export CXXINCLUDES           # The extra include paths for c++, set by the various Makefile.include files.
 export NATIVEINCLUDES        # The native include paths, set by the various native Makefile.include files.
+export DEVELHELP             # Set to 1 to enable code in RIOT that does safety checking which is not needed in a production environment but helps in the development process.
```

Comment:
Hmmm...why is this being exported?

Reply:
This is part of the first commit.  It is because the `MODULE_MPU_STACK_GUARD` depends on this so kconfing must be able to check it.

Reply:
Can you `target-export-variables` only for Kconfig targets?

Reply:
Hmmm, I wonder.  It would be needed for any dependency resolution if `TEST_KCONFIG=1`.

Reply:
I would prefer not to globally export something that does not need to be globally exported, once exported its really hard to know what are the users of the exported variable, and changes to it become harder.

Reply:
We should carefully check which of all the Kconfig targets may need it.
As an alternative I'm thinking to have an equivalent Kconfig symbol for develhelp, and set it using https://github.com/RIOT-OS/RIOT/pull/16052

Reply:
So would there be a:
```
ifeq (1, $(DEVELHELP))
  RIOT_CONFIG_DEVELHELP=y
endif
```

or something to passthrough the config?

Reply:
I don't mind getting in #16052 first as it looks about there.

Reply:
> So would there be a:

Yes, I imagine something like that, did not test it though...